File: Die Collatz Serie/Collatz_Serie.java
import java.util.Scanner;

public class Collatz {
    public static void main(String[] args) {
        Scanner num = new Scanner(System.in);
        System.out.println("Enter a number: ");
        int n = Integer.parseInt(num.nextLine());
        
        System.out.print("The collatz sequence for " + n + " is: ");
        doCollatz(n, "");
    }
    
    public static void doCollatz(int n, String sequence) {
        while (n != 1) {
            if (n % 2 == 0) {
                n /= 2;
            } else {
                n = 3 * n + 1;
            }
            sequence += n + " ";
        }
        
        System.out.println(sequence);
    }
}

File: Die Collatz Serie/Collatz_Serie.py
def collatz_conjecture(num):
   num = int(input('Enter a number: '))
   sequence = [num]
   while(num != 1):
       if((num%2)==0):
           num = num // 2 
       else:
           num = (num*3) + 1
       sequence.append(num)
   return sequence
sequence = collatz_conjecture(num=True)
print(sequence)